### PR TITLE
Svnn ibc proof generation

### PR DESCRIPTION
**Pull request by systemzax**
This PR updates the standalone svnn_ibc unit test to include proof generation. Resolves https://github.com/AntelopeIO/spring/issues/90.

### CODE REVIEW

File: unittests/test-contracts/svnn_ibc/svnn_ibc.hpp [hunk 1]
```diff
@@ -50,46 +47,43 @@ CONTRACT svnn_ibc : public contract {
...
-          return proof;
-      }
+         uint64_t layers_depth = calculate_max_depth(c_leaf_count) -1;
 
-      */
+         for (uint64_t i = 0; i < layers_depth; i++) {
+            bool isLeft = leaf_index % 2;
+            uint64_t pairIndex = isLeft ? leaf_index - 1 :
+                           leaf_index == leaf_count ? leaf_index :
+                           leaf_index + 1;
```
Comment (greg7mdp) [inline unittests/test-contracts/svnn_ibc/svnn_ibc.hpp:64]: Better use parentheses so the reader doesn't have to remember esoteric operator precedence.
```suggestion
            uint64_t pairIndex = isLeft ? leaf_index - 1 :
                           (leaf_index == leaf_count ? leaf_index : leaf_index + 1);

```

File: unittests/test-contracts/svnn_ibc/svnn_ibc.hpp [hunk 2]
```diff
@@ -133,22 +127,22 @@ CONTRACT svnn_ibc : public contract {
...
 
           uint32_t         block_num;
           
           checksum256      finality_mroot;
 
           time_point       cache_expiry;
 
-          uint64_t primary_key()const { return id; }
-          uint64_t by_block_num()const { return block_num; }
+          //uint64_t primary_key()const { return id; }
```
Comment (greg7mdp) [inline unittests/test-contracts/svnn_ibc/svnn_ibc.hpp:144]: please remove commented line.

File: unittests/test-contracts/svnn_ibc/svnn_ibc.hpp [hunk 3]
```diff
@@ -133,22 +127,22 @@ CONTRACT svnn_ibc : public contract {
...
 
-         EOSLIB_SERIALIZE( storedpolicy, (generation)(fthreshold)(finalizers)(last_block_num)(cache_expiry))
+         EOSLIB_SERIALIZE( storedpolicy, (generation)(threshold)(finalizers)(last_block_num)(cache_expiry))
 
       };
 
       TABLE lastproof {
 
-          uint64_t         id;
+          //uint64_t         id;
```
Comment (greg7mdp) [inline unittests/test-contracts/svnn_ibc/svnn_ibc.hpp:136]: please remove commented line.

File: unittests/test-contracts/svnn_ibc/svnn_ibc.hpp [hunk 4]
```diff
@@ -100,20 +94,20 @@ CONTRACT svnn_ibc : public contract {
       }
 
       struct quorum_certificate {
-          std::vector<uint64_t> finalizers;
-          bls_signature signature;
+          std::vector<uint32_t>  finalizers;
```
Comment (greg7mdp) [inline unittests/test-contracts/svnn_ibc/svnn_ibc.hpp:97]: I believe we switched to `std::vector<uint8_t> ` in spring.

File: unittests/test-contracts/svnn_ibc/svnn_ibc.hpp [hunk 5]
```diff
@@ -50,46 +47,43 @@ CONTRACT svnn_ibc : public contract {
...
+         for (uint64_t i = 0; i < layers_depth; i++) {
+            bool isLeft = leaf_index % 2;
+            uint64_t pairIndex = isLeft ? leaf_index - 1 :
+                           leaf_index == leaf_count ? leaf_index :
+                           leaf_index + 1;
+   
+            if (pairIndex<leaf_count) proof_path.push_back(isLeft);
+   
+            leaf_count/=2;
+            leaf_index=leaf_index / 2;
```
Comment (greg7mdp) [inline unittests/test-contracts/svnn_ibc/svnn_ibc.hpp:69]: why not `/=2` to be consistent with line above?

**Review by linh2931 — APPROVED** [review]

**Pull request merged**